kernel/src/common/regs/README.md

```diff
@@ -0,0 +1,207 @@
+# Register/Bitfield Interface for Tock
+
+This module provides an interface for defining and manipulating memory mapped
+registers and bitfields.
+
+## Defining registers
+
+The module provides three types for working with memory mapped registers: `ReadWrite`,
+`ReadOnly`, and `WriteOnly`, providing read-write, read-only, and write-only functionality,
+respectively.
+
+Defining the registers is similar to the C-style approach, where each register
+is a field in a packed struct:
+
+```rust
+use common::regs::{ReadOnly, ReadWrite, WriteOnly};
+
+#[repr(C, packed)]
+struct Registers {
+    // Control register: read-write
+    // The 'Control' parameter constrains this register to only use fields from
+    // a certain group (defined below in the bitfields section).
+    cr: ReadWrite<u8, Control>,
+
+    // Status register: read-only
+    s: ReadOnly<u8, Status>
+
+    // Registers can be bytes, halfwords, or words:
+    // Note that the second type parameter can be omitted, meaning that there
+    // are no bitfields defined for these registers.
+    byte0: ReadWrite<u8>,
+    byte1: ReadWrite<u8>,
+    short: ReadWrite<u16>,
+    word: ReadWrite<u32>
+
+    // Etc.
+}
+```
+
+## Defining bitfields
+
+Bitfields are defined through the `bitfields!` macro:
+
+```rust
+bitfields! [
```

Comment:
Can this macro be called multiple times in the same file?

Reply:
Yes.

Reply:
Also, what about calling the macro `registers!`?

Reply:
I sorta think that a `registers!` macro would define all the registers for a peripheral (i.e. would create the `struct Registers` object). I like bitfields. Maybe `register_bitfields!` if you want to be pedantic

Reply:
Well I could argue that the macro is being used to define registers, but yeah I kinda like `register_bitfields`. `bitfields` is pretty generic, and is pretty close to https://docs.rs/bitfield/0.12.1/bitfield/macro.bitfield.html which could confuse people.